fcrepo4/fcrepo4
# implemented new storage layout

**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3375

# What does this Pull Request do?

Updates the storage layout so that it's inline with this wiki: https://wiki.lyrasis.org/display/FF/Design+-+Fedora+OCFL+Object+Structure

# How should this be tested?

All of the existing features should work the same. I would create a variety of different types of resources, to make sure the APIs still work as expected, and then verify the OCFL representations.

The easiest, in my biased opinion, way to do this is by using [rocfl](https://crates.io/crates/rocfl). To do this, install the utility, and then navigate to your ocfl storage root. From there, you can execute commands like:

```shell
# List the objects
$ rocfl ls
info:fedora                               
info:fedora/binary                        
info:fedora/box                           
info:fedora/box2 

#List the contents of an object
$ rocfl ls info:fedora/binary
.fcrepo/fcr-root.json                     
.fcrepo/fcr-root~fcr-desc.json            
binary                                    
binary~fcr-desc.nt

#List an old version of an object
$ rocfl ls -v1 info:fedora/box2
.fcrepo/fcr-root.json                     
fcr-container.nt
```
Additional information can be displayed using the `-l` and/or `-p` flags.

# Interested parties
@fcrepo4/committers
